app/frontend/src/features/communities/events/EventForm.tsx

```diff
@@ -0,0 +1,257 @@
+  },
+  form: {
+    display: "grid",
+    rowGap: theme.spacing(3),
+    marginBlockEnd: theme.spacing(3),
+  },
+  duoContainer: {
+    display: "grid",
+    gridTemplateColumns: "1fr",
+    gap: theme.spacing(3, 2),
+    [theme.breakpoints.up("md")]: {
+      gridTemplateColumns: "1fr 1fr",
+    },
+  },
+  locationContainer: {
+    minHeight: theme.typography.pxToRem(66),
+  },
+  endDateTimeButton: {
+    justifySelf: "start",
+  },
+  isOnlineCheckbox: {
+    width: "50%",
+  },
+  eventDetailsContainer: {
+    display: "grid",
+    rowGap: theme.spacing(1),
+  },
+  createEventButton: {
+    justifySelf: "end",
+  },
+}));
+
+interface BaseEventData {
+  content: string;
+  title: string;
+  startDate: Dayjs;
+  endDate: Dayjs;
+  startTime: Dayjs;
+  endTime: Dayjs;
+  isOnline: boolean;
+  eventImage?: string;
+  parentCommunityId?: number;
+  link?: string;
+  location?: GeocodeResult;
+}
+interface OfflineEventData extends BaseEventData {
+  isOnline: false;
+  location: GeocodeResult;
+}
+
+interface OnlineEventData extends BaseEventData {
+  isOnline: true;
+  link: string;
+  parentCommunityId: number;
+}
+
+export type CreateEventData = OfflineEventData | OnlineEventData;
+
+interface EventFormProps {
+  error: GrpcError | null;
+  mutate: UseMutateFunction<unknown, GrpcError, CreateEventData, unknown>;
+  isMutationLoading: boolean;
+  isMutationSuccess: boolean;
+  successMessage: string;
+  title: string;
+}
+
+export default function EventForm({
+  error,
+  mutate,
+  isMutationLoading,
+  isMutationSuccess,
+  successMessage,
+  title,
+}: EventFormProps) {
+  const classes = useStyles();
+  const { control, errors, handleSubmit, register, watch } =
+    useForm<CreateEventData>();
+
+  const isOnline = watch("isOnline", false);
+
+  const onSubmit = handleSubmit(
+    (data) => {
+      mutate(data);
+    },
+    (errors) => {
+      if (errors.eventImage) {
+        window.scroll({ top: 0, behavior: "smooth" });
+      }
+    }
+  );
+
+  return (
+    <>
+      <ImageInput
+        alt={EVENT_IMAGE_INPUT_ALT}
+        className={classes.imageInput}
+        control={control}
+        id="event-image-input"
+        name="eventImage"
+        type="rect"
+      />
+      <PageTitle>{title}</PageTitle>
+      {error || errors.eventImage || errors.location ? (
+        <Alert severity="error">
+          {error?.message ||
+            errors.eventImage?.message ||
+            // @ts-expect-error
+            errors.location?.message ||
+            ""}
+        </Alert>
+      ) : (
+        isMutationSuccess && <Alert severity="success">{successMessage}</Alert>
+      )}
+      <form className={classes.form} onSubmit={onSubmit}>
+        <TextField
+          fullWidth
+          id="title"
+          inputRef={register({ required: TITLE_REQUIRED })}
+          name="title"
+          label={TITLE}
+          variant="standard"
+        />
+        <div className={classes.duoContainer}>
+          <Datepicker
```

Comment:
I think the default start and end date should be tomorrow (arguable), and the default time should be rounded to the nearest hour.

Also, when you change the date, the end date should also change to the same interval apart (ie. if the dates are 2021-01-01 and 2021-01-03 and the start date is changed to 2021-01-03, the end date should change to 2021-01-05).
Same for the end time.

Maybe we need to add a `onPostChange` callback to call `setValue` in?